ql/src/java/org/apache/hadoop/hive/ql/exec/tez/YarnQueueHelper.java

```diff
@@ -92,6 +94,20 @@ public YarnQueueHelper(HiveConf conf) {
   }
 
   public void checkQueueAccess(
+      String queueName, String userName) throws IOException, InterruptedException {
+    UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
```

Comment:
shouldn't we use `getLoginUser()` here?

Reply:
I had a discussion regarding this with @abstractdog and he is thinking that if the current user is changed from the login user somewhere upper in the call stack, thank we should respect that, since that change had some reason.